homeassistant/components/netatmo/sensor.py

```diff
@@ -73,23 +74,61 @@
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the available Netatmo weather sensors."""
     netatmo = hass.components.netatmo
-    data = NetAtmoData(netatmo.NETATMO_AUTH, config.get(CONF_STATION, None))
 
     dev = []
+    if CONF_MODULES in config:
+        manual_config(netatmo, config, dev)
+    else:
+        auto_config(netatmo, config, dev)
+
+    if dev:
+        add_entities(dev, True)
+
+
+def manual_config(netatmo, config, dev):
+    """Handle manual configuration."""
     import pyatmo
-    try:
-        if CONF_MODULES in config:
+
+    handled = []
+    not_handled = []
+    for data_class in [pyatmo.WeatherStationData, pyatmo.HomeCoachData]:
+        data = NetAtmoData(netatmo.NETATMO_AUTH, data_class,
+                           config.get(CONF_STATION))
+        try:
             # Iterate each module
             for module_name, monitored_conditions in \
                     config[CONF_MODULES].items():
                 # Test if module exists
                 if module_name not in data.get_module_names():
-                    _LOGGER.error('Module name: "%s" not found', module_name)
+                    if module_name in handled:
+                        # already handled by other data class
+                        continue
+                    if module_name not in not_handled:
+                        not_handled.append(module_name)
                     continue
+                else:
+                    if module_name in not_handled:
+                        not_handled.remove(module_name)
+                    if module_name not in handled:
+                        handled.append(module_name)
```

Comment:
These nested `if`s are quite complex. Could something like this work instead?

```python
all_classes = [pyatmo.WeatherStationData, pyatmo.HomeCoachData]
not_handled = {}
# ...
if module_name not in data.get_module_names():
    not_handled[module_name] = not_handled[module_name]+1 if module_name in not_handled else 1
# ...
for module_name in [m for m in not_handled if not_handled[m]==len(all_classes)]:
    _LOGGER.error('Module name: "%s" not found', module_name)
```

Reply:
Applied your suggestion